sdk/communication/AzureCommunicationChat/Source/AzureCommunicationChatClient.swift

```diff
@@ -109,7 +109,7 @@ public final class AzureCommunicationChatClient: PipelineClient, PageableClient
         }
 
         // Send request
-        let context = PipelineContext.of(keyValues: [
+        let context = options?.context ?? PipelineContext.of(keyValues: [
```

Comment:
Should we be merging in the "default" context entries here rather than just taking the user's context object as-is? I can see arguments both ways - the user shouldn't need to know to set the allowed status codes for example, but also if for some reason the service changes and a new status code is added the user needs an escape hatch to work around our default value for that context key.

Reply:
If we update the default context with any user keys, they should be able to do both.

Reply:
@bsiegel Is it possible to offer both public APIs? (i.e. something like `replaceContext` and `mergeContext`).

Reply:
Sounds like the answer is that we just need essentially, `mergeContext`